src/test/unit/utils/SerializationSTLTest.cpp

```diff
@@ -0,0 +1,59 @@
+#include <gtest/gtest.h>
+
+#include <nupic/types/Serializable.hpp>
+#include <nupic/utils/StlIo.hpp>
+#include <sstream>
+
+#include <vector>
+
+namespace testing {
+using namespace std;
+using namespace nupic;
+class Dummy : public Serializable {
+  public:	
+  int i = 3;
+  vector<int> v = vector<int>{1, 2, 3};
+
+  //Serializable interface
+  void save(ostream &o) const override {
+    o << "DUMMY ";
+    o << i << " ";
+//!    o << v << " ";
```

Comment:
Yes, this is a good objective.  
Actually, the vector binary serialization did work before but the logic was hard to follow.

The best way to do this is to provide binary serializations for each of the STL classes that use save( ) and load( ) and then make another set of human readable serializations for each of the STL classes which will use the  << and >> operators.

Remember that the STL classes could be nested.  How do we serialize something like `vector<std::deque<std::map<std::string>>>`  ?

Now, this is a common problem.  I wonder if someone already did this someplace on the web?

Reply:
> emember that the STL classes could be nested. How do we serialize something like vector<std::deque<std::map<std::string>>> ?

this is why I wanted to go with the binary serialization, I had hoped all STL Containers use continuous memory, so the binary version would just dump/load whatever is inside. 

>  That works because the vector's buffer is contiguous. But that does not work with std::set or std::map etc. because each element's pair is allocated independently. ~DK

But you say it's not true (only for vectors). 

> Yes, this is a good objective.

this is why I wanted to override `operator<<`, as it's used only at a few places for "print", and print() could be used there. While when writing serialization, we stream all primitive types with `<<`, so I wanted to provide that for STL too.